### ISSUE TITLE
Fig prevents git operations in VS Code from working

### ISSUE THREAD

**Issue by imgrant** [withfig/fig#1875]
### Sanity checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my issue
- [X] I have run `fig doctor` in the affected terminal session
- [X] I have run `fig restart` and tested again (tell us if that fixed it)

### Description:

Using the [GitLens plugin](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) with VS Code, operations via the UI such as delete branch do not work if there is not already a GitLens terminal window open in VS Code.

### Steps to reproduce:

1. Install/be using VS Code with the GitLens plugin; ensure no (GitLens) terminal windows are open in VS Code.
2. Delete a branch using the source control UI. e.g. right click the branch name and choose 'Delete branch'
![image](https://user-images.githubusercontent.com/5815412/193863854-0fc85b4d-0f93-413e-b434-f875aaaa9e8c.png)
3. Observe a GitLens terminal window is opened in VS Code (to run the git delete operation), but that the branch is *not* deleted. E.g. the GitLens terminal window shows:
```
cd "/Users/user/repo/branch"
git branch --delete test
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish

$
```
4. Repeat the delete branch operation (leaving the GitLens terminal window open). Observe the branch is deleted, and the operation shown in the existing GitLens terminal window:
```
git branch --delete test
Deleted branch test (was c5a409af3).
```
5. Note, if you closed the GitLens terminal window before step 4, the result is as from step 3.

### Expectation:

The branch should be deleted on the first try.

### Diagnostics:

If I remove the files `~/.config/fish/conf.d/00_fig_pre.fish` and `~/.config/fish/conf.d/99_fig_post.fish`, then it works as expected.
This is probably the same cause behind #1874 (fig startup interrupts the shell in some way).

### Environment

```yaml
fig-details:
  - desktop-version: Version 1.0.61 (B538) [Beta] [British]
hardware-info:
  - model: MacBook Pro
  - model-id: MacBookPro18,2
  - chip-id: Apple M1 Max
  - cores: 10 (8 performance and 2 efficiency)
  - mem: 64 GB
os-info:
  - macOS 12.6.0 (21G115)
environment:
  - shell: /nix/store/lwaly4qzlf9vmx5jrag1zrzc2df4bpdd-fish-3.4.0/bin/fish
  - terminal: iterm
  - cwd: /Users/ian/.config/fish/conf.d
  - exe-path: /Users/ian/.local/bin/fig
  - install-method: brew
  - env-vars:
    - PATH: /Users/ian/.pyenv/shims:/Users/ian/.pyenv/bin:/Users/ian/.nix-profile/bin:/nix/var/nix/profiles/default/bin:/Users/ian/.config/composer/vendor/bin:/Users/ian/go/bin:/Users/ian/.local/bin:/Users/ian/.fig/bin:/opt/local/bin:/opt/local/sbin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Applications/VMware Fusion Tech Preview.app/Contents/Public:/usr/local/MacGPG2/bin:/Users/ian/.nix-profile/bin:/etc/profiles/per-user/ian/bin:/run/current-system/sw/bin:/nix/var/nix/profiles/default/bin
    - SHELL: /run/current-system/sw/bin/fish
    - TERM_SESSION_ID: e154a144-fc63-4427-9334-00ac0cbc2a64
    - FIG_SET_PARENT: e154a144-fc63-4427-9334-00ac0cbc2a64
    - FIG_TERM_VERSION: 5.2.0
    - FIG_TERM: 1
    - FIG_PARENT:
    - FIG_INTEGRATION_VERSION: 8
    - TERM: xterm-256color
    - FIG_PID: 1095
- integrations:
  - SSH: false
  - TMUX: false
  - iTerm: installed!
  - Hyper: application is not present.
  - Visual Studio Code: installed!
  - Docker: true
```

**Comment by squili** [comment]
Could you check if this is still happening on the latest version of Fig?

**Comment by imgrant** [comment]
> Could you check if this is still happening on the latest version of Fig?

Yes, it still happens unfortunately. I am actually on a new Mac since I reported this, so I have had the luxury of setting it up again. Hence, I just installed the latest Fig, added fish (so a clean fish config), and it still happens.

I was further able to test, removing just the pre file, `~/.config/fish/conf.d/00_fig_pre.fish`, fixes the issue (i.e. post file is fine). Additionally, it doesn't happen with zsh.
